lib/spack/docs/environments.rst

```diff
@@ -368,6 +368,34 @@ from being added again.  At the same time, a spec that already exists in the
 environment, but only as a dependency, will be added to the environment as a
 root spec without the ``--no-add`` option.
 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Developing Packages in a Spack Environment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``spack develop`` command allows one to develop Spack packages in
+an environment. It requires a spec containing a concrete version, and
+will configure Spack to install the package from local source. By
+default, it will also clone the package to a subdirectory in the
+environment. This package will have a special variant ``develop_path``
+set, and Spack will ensure the package and its dependents are rebuilt
+any time the environment is installed if the package's local source
+code has been modified. Spack ensures that all instances of a
+developed package in the environment are concretized to match the
+version (and other constraints) passed as the spec argument to the
+``spack develop`` command.
+
+For packages with ``git`` attributes, 40-character commit hashes are
+also valid versions. This means that for a package ``foo`` whose
+latest commit is ``abcdef1234abcdef1234abcdef1234abcdef1234``, ``spack
+develop foo@abcdef1234abcdef1234abcdef1234abcdef1234`` will clone the
+latest version of the package, and ``spack install`` will install from
```

Comment:
so "latest version" in this case follows the rules below? `develop > main> master>trunk`. or is it going to follow the tag check and clone off the "closest version" as outlined in the `Version` class documentation? for a develop spec I think the former would be appropriate.  It might be good to clarify that here and/or provide a direct link to the `Version` source documentation.

Reply:
In this case, it's the "latest" because in the previous clause, I mentioned that this happens to be the latest commit of the package, and we know that as developers of the package. In general, a commit version would slot after `develop/main/master/head/trunk` and ahead of any known versions whose git tags are ancestors of that commit.